reconcile/utils/terrascript_client.py

```diff
@@ -3179,6 +3345,22 @@ def populate_tf_resource_elasticsearch(self, resource, namespace_info):
         es_values["elasticsearch_version"] = \
             values.get('elasticsearch_version')
 
+        log_group_resources, publishing_options = \
+            self._get_tf_resource_elasticsearch_log_groups(
+                identifier=identifier,
+                account=account,
+                resource=resource,
+                values=values,
+                output_prefix=output_prefix
+            )
+        tf_resources += log_group_resources
+
+        resource_policy = self._get_tf_resource_elasticsearch_resource_policy(
```

Comment:
are we actually creating this policy only once per account or once per account per usage of the elasticsearch provider?

either way, i think terraform covers us and will only create a resource once, even if it is defined multiple times.

maybe i'm just not reading this right?

Reply:
in addition, would it make sense to call `_get_tf_resource_elasticsearch_resource_policy` from within `_get_tf_resource_elasticsearch_log_groups`?

Reply:
Tbh I do not see a reason to call `_get_tf_resource_elasticsearch_resource_policy` from within `_get_tf_resource_elasticsearch_log_groups`, because:

- `_get_tf_resource_elasticsearch_log_groups` returns the log_groups of that specific terraform resource. 
- `_get_tf_resource_elasticsearch_resource_policy` returns a policy for the whole AWS account.

This is not well reflected on the naming though. Will rename the function to `_get_elasticsearch_account_wide_resource_policy` to highlight that better.

Reply:
> are we actually creating this policy only once per account or once per account per usage of the elasticsearch provider?

To make sure we do not hit any limits, I suggest to have a single resource policy per account for the log-groups. In the current implementation, `_get_tf_resource_elasticsearch_resource_policy` is called on every ES terraform resource. The policy name is static though, which also makes its terraform identifier static:

```
'policy_name': 'es-log-publishing-permissions'
```

By doing so, the resource appears only once in the final terraform `config.json`.

Still, this is kind of hacky as we redundantly re-compute the policy for every ES resource. It would be nice to do it in a single run, but I am currently not aware of a smoother approach.

Reply:
i believe this actually happens a lot throughout our code. you can proceed with that from my perspective!